plugins/modules/web_app_pool_info.ps1

```diff
@@ -0,0 +1,96 @@
+#!powershell
+
+# Copyright: (c) 2024, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+    options = @{
+        name = @{ type = "str" }
+    }
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$name = $module.Params.name
+
+$module.Result.exists = $false
+$module.Result.appPools = @()
+
+
+# Ensure WebAdministration module is loaded
+if ($null -eq (Get-Module -Name "WebAdministration" -ErrorAction SilentlyContinue)) {
+    Import-Module WebAdministration
+    $web_admin_dll_path = Join-Path $env:SystemRoot system32\inetsrv\Microsoft.Web.Administration.dll
+    Add-Type -LiteralPath $web_admin_dll_path
+}
+
+function Get-AppPoolInfo ($name) {
+
+    # Get all the current attributes for the pool
+    $pool = Get-Item -LiteralPath IIS:\AppPools\$name
+    if ($null -ne $pool) {
+        $module.Result.exists = $true
+    }
+    $appPoolInfoDict = @{
+        name = $name
+        info = @{}
+    }
+
+    $elements = @("attributes", "cpu", "failure", "processModel", "recycling")
```

Comment:
Is there a reason why `attributes` is separate? We try and make the return value structure similar to the non info module so we should have `$appPoolInfoDict = @{ name = $name; attributes = @{} }` and the attributes returned in a similar format to how people would specify `microsoft.iis.web_app_pool: attribute=...`. Keeping them in a similar format helps users to understand how they can set the attribute based on the info returned either for an the same pool or another one.

Reply:
To be honest the reason for that is that I aligned with the way it was implemented originally.
I don't see an issue to change it the way to suggest,
Just to be clear, the fixed format print will be something as follows:
{
    "name": "name1"
    "attributes": {
        "CLRConfigFile": "",
        "autoStart": true,
        "enable32BitAppOnWin64": false,
        "enableConfigurationOverride": true,
        "managedPipelineMode": "Classic",
        "managedRuntimeLoader": "webengine4.dll",
        ....
        ....
 
        "cpu": {
            ......
            ....
        }
        "failure": {
        .....
        ....
        },
        .......
    }
   
   
   
   So basically, include keys like cpu, failure , processModel, etc.. under attributes too.

Reply:
Implemented in
https://github.com/ansible-collections/microsoft.iis/commit/49e95c9684ce9ce31dba2a03438eda006795c927